### PR TITLE
Feature/abd hook up be to profile

### LINKED ISSUE #61: Add conditional rendering to Profile component
### Feature Categories:

- [ ] Express
- [X] React
- [ ] Styling/CSS
- [ ] Documentation
- [ ] Other: 

**Describe the feature/Bug your gonna work on**
- Currently profile information for profile pic, address, & shipping address are hard-coded
- Once back-end has edited database to include all needed fields for the user & that info gets returned on login, swap out hard-coded information
-Each should have conditional rendering to display info from store, or if there is none, a default value

### PR DESCRIPTION
**Pull request by Abdeboskey**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
* Update fields in `login` action to include new user details from DB.
* Add conditional rendering to `Profile.js` to display user details, or placeholder text if user details have not yet been submitted.
* Add default placeholder image if user has not yet uploaded picture.
* Add fun image to `NotFound.js`

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:

* This was tested in the browser with the 👀

# Message for reviewer:

* This was just an idea for a placeholder image! If you have another image in mind we'd love to talk about it.
# Checklist:
- [ ] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

Screen Shot 2020-10-15 at 4 24 49 PM

Screen Shot 2020-10-15 at 4 42 31 PM

### CODE REVIEW

File: code/web/src/modules/user/api/actions.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ export function login(userCredentials, isLoading = true) {
     return axios.post(routeApi, query({
       operation: 'userLogin',
       variables: userCredentials,
-      fields: ['user {name, email, role}', 'token']
+      fields: ['user {name, email, role, bio, image, street, city, state, zip}', 'token']
```
Comment (rwilliams659) [inline code/web/src/modules/user/api/actions.js:39]: I love how easy it was to get this set up!

Comment (JoshSevy) [inline code/web/src/modules/user/api/actions.js:39]: Thanks back end team this integration went super smoothly great work!

File: code/web/src/modules/user/Profile.js [hunk 2]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
-      <GridCell style={{ padding: '2em', textAlign: 'center' }}>
-        <img
-          src={`${APP_URL}//images/stock/men/4.jpg`}
-          style={{ 
-            marginBottom: '2em', 
-            height: '24em', 
-            boxShadow: level1, 
-            borderRadius: '10px'
+const Profile = ({ user, logout }) => {
+  const { details } = user
```
Comment (rwilliams659) [inline code/web/src/modules/user/Profile.js:22]: Good call to destructure this up here to simplify the variables being used below

File: code/web/src/modules/user/Profile.js [hunk 3]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
+          </Button>
+        </GridCell>
+      </Grid>
 
-        <H6 style={{ marginBottom: '0.5em' }}>Account email:</H6>
-        <p style={{ color: grey2, marginBottom: '2em' }}>{props.user.details.email}</p>
+      <Grid>
+        <GridCell style={{ padding: "2em", textAlign: "center" }}>
+          <img
+            src={details.image || `${APP_URL}//images/stock/men/4.jpg`}
```
Comment (rwilliams659) [inline code/web/src/modules/user/Profile.js:60]: Love the default img 🥇

File: code/web/src/modules/user/Profile.js [hunk 4]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
+            Logout
+          </Button>
+        </GridCell>
+      </Grid>
 
-        <H6 style={{ marginBottom: '0.5em' }}>Account email:</H6>
-        <p style={{ color: grey2, marginBottom: '2em' }}>{props.user.details.email}</p>
+      <Grid>
+        <GridCell style={{ padding: "2em", textAlign: "center" }}>
+          <img
```
Comment (JoshSevy) [inline code/web/src/modules/user/Profile.js:59]: Great work Team! So smoothe how this all worked out. Good planning went a long way. So excited.

**Review by rwilliams659 — APPROVED** [review]
Great job team, glad we got this hooked up so fast!

**Review by JoshSevy — APPROVED** [review]
Great work everyone! MVP here we come.

**Review by mlynch5187 — APPROVED** [review]

File: code/web/src/modules/user/Profile.js [hunk 5]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
-            height: '24em', 
-            boxShadow: level1, 
-            borderRadius: '10px'
+const Profile = ({ user, logout }) => {
+  const { details } = user
+
+  return (
+    <div>
+      {/* SEO */}
+      <Helmet>
```
Comment (Strompy) [inline code/web/src/modules/user/Profile.js:27]: Is this Search Engine Optimization? Does the html play a role in that or is it just the content?

File: code/web/src/modules/user/Profile.js (hunk 3 above)
Comment (Strompy) [reply code/web/src/modules/user/Profile.js:60]: Excited to see this in production

File: code/web/src/modules/user/api/actions.js (hunk 1 above)
Comment (Strompy) [reply code/web/src/modules/user/api/actions.js:39]: Glad this all came together easily!

**Review by Strompy — APPROVED** [review]

File: code/web/src/modules/user/Profile.js (hunk 5 above)
Comment (Abdeboskey) [reply code/web/src/modules/user/Profile.js:27]: The `<head>` element in an HTML document typically contains a whole bunch of metadata that is used for SEO as well as things like the title of the page that shows up in the browser tab and other fun stuff.

Comment (Abdeboskey) [reply code/web/src/modules/user/Profile.js:27]: This helmet component is just updating what is in the `<head>` as far as I understand

File: code/web/src/modules/common/NotFound.js [hunk 6]
```diff
@@ -30,10 +30,10 @@ const NotFound = () => (
     <Grid>
       <GridCell style={{ textAlign: 'center' }}>
         <p style={{ textAlign: 'center', marginTop: '2em', marginBottom: '2em' }}>
-          <img src={`${ APP_URL }/images/crate-broken.png`} alt="404" style={{ width: '10em' }}/>
+          <img src={`${ APP_URL }/images/crate-broken.png`} alt="404" style={{ width: '25em', borderRadius: '50%' }}/>
         </p>
 
-        <H3 font="secondary">Page you are looking for does not exists or has removed. It's 404.</H3>
+        <H3 font="secondary">Page you are looking for does not exists or has been removed. It's 404.</H3>
```
Comment (sienna-kopf) [inline code/web/src/modules/common/NotFound.js:36]: Live this 404 page catch!!!

File: code/web/src/modules/user/Profile.js (hunk 2 above)
Comment (sienna-kopf) [reply code/web/src/modules/user/Profile.js:22]: I really appreciated the conversation about the de-structuring of props into individual attributes

File: code/web/src/modules/user/Profile.js [hunk 7]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
+        <title>My Profile - Crate</title>
+      </Helmet>
+
+      {/* Top title bar */}
+      <Grid style={{ backgroundColor: grey }} justifyRight={true}>
```
Comment (sienna-kopf) [inline code/web/src/modules/user/Profile.js:32]: What is the point of the double wrapped object notation here around the styling element?

File: code/web/src/modules/user/Profile.js (hunk 3 above)
Comment (sienna-kopf) [reply code/web/src/modules/user/Profile.js:60]: OOH yes! Can't wait to see!

File: code/web/src/modules/user/Profile.js [hunk 8]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
 
-        <Link to={userRoutes.editProfile.path}>
-          <Button theme="primary">Edit Profile Info</Button>
-        </Link>
-      </GridCell>
+          <H6 style={{ marginBottom: "0.5em" }}>Shipping Address:</H6>
+          <p style={{ color: grey2, marginBottom: "2em" }}>
+            {details.street && 
+              <>
+                {details.street}
```
Comment (sienna-kopf) [inline code/web/src/modules/user/Profile.js:85]: I'm glad we split these up for you here on the FE. This looks like its smooth to access.

File: code/web/src/modules/user/Profile.js [hunk 9]
```diff
@@ -18,82 +18,120 @@ import { logout } from './api/actions'
...
-
-        <img src={`${APP_URL}/images/crate.png`} style={{ height: '12em', display: 'block', margin: '2.4em auto 1em' }}/>
-        <Button theme="primary">View Order History</Button>
-      </GridCell>
-    </Grid>
-  </div>
-)
+          <H6 style={{ marginBottom: "0.5em" }}>Bio:</H6>
+          <p style={{ color: grey2, width: "60%", margin: "0 auto 2em" }}>
+            {details.bio ? `"${details.bio}"` : "You have not written a bio"}
```
Comment (sienna-kopf) [inline code/web/src/modules/user/Profile.js:97]: Very cool that you can include text here as a filler for the field in the edit form.

File: code/web/src/modules/user/api/actions.js (hunk 1 above)
Comment (sienna-kopf) [reply code/web/src/modules/user/api/actions.js:39]: This makes so much sense! I didn't do much digging into how these would connect, but this is great!

**Review by sienna-kopf — APPROVED** [review]

**Pull request merged**